x/epochs/types/hooks.go

```diff
@@ -1,14 +1,12 @@
 package types
 
-import (
-	sdk "github.com/cosmos/cosmos-sdk/types"
-)
+import context "context"
 
 type EpochHooks interface {
 	// the first block whose timestamp is after the duration is counted as the end of the epoch
-	AfterEpochEnd(ctx sdk.Context, epochInfo EpochInfo)
+	AfterEpochEnd(ctx context.Context, epochInfo EpochInfo)
 	// new epoch is next block of epoch end block
-	BeforeEpochStart(ctx sdk.Context, epochInfo EpochInfo)
+	BeforeEpochStart(ctx context.Context, epochInfo EpochInfo)
```

Comment:
why did this have to change if it was in our own module?

Reply:
not mandatory, i did it for consistency with the sdk

Reply:
`sdk.Context` now implements `context.Context` so we can pass `sdk.Context` to sdk functions and unwrap does nothing if it's already `sdk.Context`

Reply:
i'll follow your lead here, i don't have a strong preference but feel like it's the same [argument you made](https://github.com/Stride-Labs/stride/pull/1381#discussion_r2004087790) about `NewContextLegacy()` vs `NewContext()`